services/frontend-service/src/ui/Batch.test.tsx

```diff
@@ -21,18 +21,120 @@ import { Spy } from 'spy4js';
 import { BatchAction, Lock, LockBehavior } from '../api/api';
 import { ActionsCartContext } from './App';
 
-const ChildButton = (props: { state?: string; openDialog?: () => void }) => {
-    const { openDialog } = props;
+const mock_setActions = Spy('setActions');
+const finallySpy = Spy('.finally');
+
+const sampleDeployAction: BatchAction = {
+    action: {
+        $case: 'deploy',
+        deploy: {
+            application: 'dummy application',
+            version: 22,
+            environment: 'dummy environment',
+            ignoreAllLocks: false,
+            lockBehavior: LockBehavior.Ignore,
+        },
+    },
+};
+
+const newVersionDeploy: BatchAction = {
+    action: {
+        $case: 'deploy',
+        deploy: {
+            application: 'dummy application',
+            version: 30,
+            environment: 'dummy environment',
+            ignoreAllLocks: false,
+            lockBehavior: LockBehavior.Ignore,
+        },
+    },
+};
+
+const sampleUndeployAction: BatchAction = {
+    action: {
+        $case: 'undeploy',
+        undeploy: {
+            application: 'dummy application',
+        },
+    },
+};
+
+const sampleCreateEnvLock: BatchAction = {
+    action: {
+        $case: 'createEnvironmentLock',
+        createEnvironmentLock: {
+            environment: 'dummy environment',
+            lockId: '1234',
+            message: 'hello',
+        },
+    },
+};
+
+const newIdEnvLock: BatchAction = {
```

Comment:
please rename to be more in line with the other envlock, e.g. `sampleCreateEnvLockOtherId`